File: examples/word_evans.rb
#
# jruby examples for jas.
# $Id$
#

#load "examples/jas.rb"
require "examples/jas"

# non-commutative polynomial examples: evans example

r = WordPolyRing.new(QQ(),"x,y,z");
puts "WordPolyRing: " + str(r);
puts;

one,x,y,z = r.gens();
puts "one = " + str(one);
puts "x = " + str(x);
puts "y = " + str(y);
puts "z = " + str(z);
puts;

f1 = x * y - z;
f2 = y * z + 2 * x + z;
f3 = y * z + x;

puts "f1 = " + str(f1);
puts "f2 = " + str(f2);
puts "f3 = " + str(f3);
puts;

ff = r.ideal( "", [f1,f2,f3] );
puts "ff = " + str(ff);
puts;

startLog();

gg = ff.GB();
puts "gg = " + str(gg);
puts "isGB(gg) = " + str(gg.isGB());
puts;

#exit(0);

c1 = f1 * f2;
c2 = f2 * f1;
s = c1 - c2;

puts "c1 = " + str(c1);
puts "c2 = " + str(c2);
puts "s  = " + str(s);
puts;

ff = r.ideal( "", [f1,f2,f3,c1,c2,s] );
puts "ff = " + str(ff);
puts;

gga = ff.GB();
puts "gga = " + str(gga);
puts "isGB(gga) = " + str(gga.isGB());
puts "gg == gga: " + str(gg.list == gga.list);
puts;

File: examples/word_evans.py
#
# jython examples for jas.
# $Id$
#

#from java.lang import System

from jas import WordRing, WordPolyRing, WordPolyIdeal
from jas import terminate, startLog
from jas import QQ, ZZ, GF, ZM

# non-commutative polynomial examples: evans example

r = WordPolyRing(QQ(),"x,y,z");
print "WordPolyRing: " + str(r);
print;

[one,x,y,z] = r.gens();
print "one = " + str(one);
print "x = " + str(x);
print "y = " + str(y);
print "z = " + str(z);
print;

f1 = x * y - z;
f2 = y * z + 2 * x + z;
f3 = y * z + x;

print "f1 = " + str(f1);
print "f2 = " + str(f2);
print "f3 = " + str(f3);
print;

F = r.ideal( list=[f1,f2,f3] );
print "F = " + str(F);
print;

startLog();

G = F.GB();
print "G = " + str(G);
print "isGB(G) = " + str(G.isGB());
print;

#exit(0);

c1 = f1 * f2;
c2 = f2 * f1;
s = c1 - c2;

print "c1 = " + str(c1);
print "c2 = " + str(c2);
print "s  = " + str(s);
print;

Fa = r.ideal( list=[f1,f2,f3,c1,c2,s] );
print "Fa = " + str(Fa);
print;

Ga = Fa.GB();
print "Ga = " + str(Ga);
print "isGB(Ga) = " + str(Ga.isGB());
#print;

#print "G = " + str(G);
print "G == Ga: " + str(cmp(G,Ga));
print "G == Ga: " + str(G.list == Ga.list);
print;

